core/src/main/scala/akka/kafka/internal/ProducerStage.scala

```diff
@@ -19,7 +21,8 @@ import java.util.concurrent.atomic.AtomicInteger
  * INTERNAL API
  */
 private[kafka] class ProducerStage[K, V, P](
-  settings: ProducerSettings[K, V], producerProvider: () => KafkaProducer[K, V]
+  closeTimeout: FiniteDuration,
+  producerProvider: () => KafkaProducer[K, V], closeProducerOnStop: Boolean
```

Comment:
minor subjective nitpick: I'd prefer to have `closeProducerOnStop` after `closeTimeout`